include/deal.II/particles/particle_accessor.h

```diff
@@ -680,6 +703,25 @@ namespace Particles
 
 
 
+  template <int dim, int spacedim>
+  inline void
+  ParticleAccessor<dim, spacedim>::set_properties(
+    const Tensor<1, dim> &new_properties)
+  {
+    Assert(state() == IteratorState::valid, ExcInternalError());
+
+    // A Tensor object is not an array, so we cannot just create an
+    // ArrayView object for it. Rather, copy the data into a true
+    // array and make the ArrayView from that.
+    double array[dim];
+    for (unsigned int d = 0; d < dim; ++d)
+      array[d] = new_properties[d];
+
+    set_properties(make_array_view(array));
```

Comment:
This is where I would like to do
```suggestion
    set_properties(make_array_view(new_properties.to_array()));
```
as mentioned in the issue.

Reply:
Come to think of it - if we *do* implement `to_array()` then we don't need this extra function at all, so we should just implement `to_array()` :)

Reply:
The constructor is not `explicit`:
```
  template <std::size_t N>
  ArrayView(value_type (&array)[N]);
```

I can't quite figure out from your comments where you are asking me to make changes?

Reply:
I'll put up a PR with what I want in it.

Reply:
How about we merge this now and then do what you suggest as a follow-up? It doesn't become easier/less work for anyone when we try to interleave patches from different people :-)